plugins/feedback/src/components/FeedbackDetailsModal/FeedbackDetailsModal.tsx

```diff
@@ -0,0 +1,356 @@
+  }>({ status: null, assignee: null, avatarUrls: null, element: null });
+
+  const [isLoading, setIsLoading] = useState(true);
+  const [expandDescription, setExpandDescription] = useState(false);
+
+  useEffect(() => {
+    if (modalData?.ticketUrl) {
+      api
+        .getTicketDetails(
+          modalData.feedbackId,
+          modalData.ticketUrl,
+          modalData.projectId,
+        )
+        .then(data => {
+          setTicketDetails({
+            status: data.status,
+            assignee: data.assignee,
+            avatarUrls: data.avatarUrls,
+            element: (
+              <>
+                <ListItem>
+                  <ListItemText primary="Status" />
+                  <ListItemSecondaryAction>
+                    <ListItemText primary={<Chip label={data.status} />} />
+                  </ListItemSecondaryAction>
+                </ListItem>
+                <ListItem>
+                  <ListItemText primary="Assignee" />
+                  <ListItemSecondaryAction>
+                    <ListItemText
+                      primary={
+                        <Chip
+                          avatar={
+                            <Avatar
+                              src={
+                                data.avatarUrls ? data.avatarUrls['48x48'] : ''
+                              }
+                            />
+                          }
+                          label={data.assignee ? data.assignee : 'Unassigned'}
+                        />
+                      }
+                    />
+                  </ListItemSecondaryAction>
+                </ListItem>
+              </>
+            ),
+          });
+          setIsLoading(false);
+        });
+    } else {
+      setIsLoading(false);
+    }
+    if (!modalData && queryState) {
+      api.getFeedbackById(queryState).then(resp => {
+        if (resp?.error !== undefined) {
+          setSnackbarOpen({ message: resp.error, open: true });
+        } else {
+          const respData: FeedbackType = resp?.data!;
+          setModalData(respData);
+        }
+      });
+    }
+  }, [modalData, api, queryState]);
+
+  const handleSnackbarClose = (
+    event?: React.SyntheticEvent,
+    reason?: string,
+  ) => {
+    event?.preventDefault();
+    if (reason === 'clickaway') {
+      return;
+    }
+    setSnackbarOpen({ ...snackbarOpen, open: false });
+    setQueryState(undefined);
+  };
+
+  const handleClose = () => {
+    setModalData(undefined);
+    setTicketDetails({
+      status: null,
+      assignee: null,
+      avatarUrls: null,
+      element: null,
+    });
+    setIsLoading(true);
+    setExpandDescription(false);
+    setQueryState(undefined);
+  };
+
+  const getDescription = (str: string) => {
+    if (!expandDescription) {
+      if (str.length > 400) {
+        if (str.split(' ').length > 1)
+          return `${str.substring(0, str.lastIndexOf(' ', 400))}...`;
+        return `${str.slice(0, 400)}...`;
```

Comment:
Similar question here, this is getting trimmed to avoid large text strings which is good, however is this full string available to the user somewhere in the app for the off chance when it could have information in it?

Reply:
For long descriptions I have added read more button, to expand description of feedback

![image](https://github.com/janus-idp/backstage-plugins/assets/42431299/2a83b066-4aea-4cb1-ba2f-a98c4d5400f6)